staging/src/k8s.io/apiserver/pkg/server/egressselector/egress_selector.go

```diff
@@ -36,13 +36,18 @@ import (
 	utilnet "k8s.io/apimachinery/pkg/util/net"
 	"k8s.io/apiserver/pkg/apis/apiserver"
 	egressmetrics "k8s.io/apiserver/pkg/server/egressselector/metrics"
+	compbasemetrics "k8s.io/component-base/metrics"
 	"k8s.io/component-base/tracing"
 	"k8s.io/klog/v2"
 	client "sigs.k8s.io/apiserver-network-proxy/konnectivity-client/pkg/client"
 )
 
 var directDialer utilnet.DialFunc = http.DefaultTransport.(*http.Transport).DialContext
 
+func init() {
```

Comment:
should this only be done if we're actually using `sigs.k8s.io/apiserver-network-proxy/konnectivity-client/pkg/client`?

for example, in a package-level sync.Once called from `case apiserver.ProtocolHTTPConnect:` and `case apiserver.ProtocolGRPC:` in `connectionToDialerCreator`?

Reply:
My sense from @logicalhan is that static provides for better analysis. Also FWIW in my testing (scraping the metrics endpoint) these are not emitted until they are actually incremented / used, so it seems not so bad.

Reply:
> in my testing (scraping the metrics endpoint) these are not emitted until they are actually incremented / used, so it seems not so bad.

if we can rely on that behavior (@logicalhan to confirm?), then the unconditional addition seems fine

Reply:
I can confirm that metrics are lazily instantiated.